Task 1 pull request

Forthey opened the pull request.

Forthey committed:
...

Forthey committed:
Add project for first task

Forthey committed:
...

Forthey committed:
Add working solving functions

Forthey committed:
Add a = 0 case and Make input more stable to wrong data

Forthey committed:
Add README for first task

Kauter1989 commented on README.md:
```diff
@@ -1 +1 @@
-<h1>Лабы по алгосам</h1>
+<h1>Лабы по алгосам, 4 семестр</h1>
```
Какие "алгосы", либо ридми пустой, либо там что-то вразумительное

Kauter1989 commented on documents/2_1_1_Дрекалов.docx:
Эти доки какое отношение к работе имеют?

Forthey replied:
> Эти доки какое отношение к работе имеют? 

Никакое, просто неудачно в начале ветку создал, поэтому информация об их удалении осталась даже здесь...

Kauter1989 commented on task1/source/main.cpp:
```diff
@@ -0,0 +1,81 @@
+				continue;
+			}
+
+			if (a == 0 && b == 0 && c == 0) {
```
Не стоит сравнивать через == вещ числа с 0

Kauter1989 commented on task1/source/main.cpp:
```diff
@@ -0,0 +1,81 @@
+			Alg::QuadraticEq equation(a, b, c);
+			std::vector<double> const& roots = equation.solve();
+
+			std::cout << "Equation " + equation.toString() + " has " << roots.size() << " real roots" << std::endl;
```
Вывод не выведен в ф-ю, вся работа с cin/cout выкинута в main

Forthey replied:
> Вывод не выведен в ф-ю, вся работа с cin/cout выкинута в main @Kauter1989 

Разве вывод в функции не вредит универсальности класса решения квадратного уравнения? Я просто специально всю работу с консолью отделил от класса квадратного уравнения, чтобы можно было, например, поменять ввод/вывод на файлы, не меняя класс

Kauter1989 commented on task1/source/main.cpp:
```diff
@@ -0,0 +1,81 @@
+
+int main() {
+	Program program;
+	program.start();
```
start и Program - абсолютно не говорящие ничего названия, названия переменных и классов должны отражать их суть

Kauter1989 commented on task1/source/quadratic_equation.cpp:
```diff
@@ -0,0 +1,49 @@
+
+void QuadraticEq::solveAsLinear()
+{
+	if (b != 0) {
```
сравнение с 0 некорректно

Kauter1989 commented on task1/source/quadratic_equation.cpp:
```diff
@@ -0,0 +1,49 @@
+
+void Alg::QuadraticEq::solveAsQuadratic()
+{
+	double D = b * b - 4 * a * c;
```
вещ числа и целочисленные константы не должны быть в выражении, не выключайте такие ворнинги, их надо избегать в коде непосредственно

Forthey committed:
Update README.md

Forthey committed:
Update README.md

Forthey committed:
Update README.md

Forthey committed:
Fix some minore issues and add more stability

As answers to [this](https://stackoverflow.com/questions/19837576/comparing-floating-point-number-to-zero) question said, comparing some number and 0 means that I should just compare it to +/-0.0
After it was checked by myself that -0.0 and +0.0 are equal, so
comparison a == 0.0 should be enough

Forthey committed:
Add some const qualifiers
